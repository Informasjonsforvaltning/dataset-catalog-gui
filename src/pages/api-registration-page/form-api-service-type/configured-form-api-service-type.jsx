import { reduxForm } from 'redux-form';
import { FormApiServiceType } from './form-api-service-type.component';
import { asyncValidate } from '../async-patch/async-patch';

const config = {
  form: 'apiServiceType',
  asyncValidate
};

export const ConfiguredFormApiServiceType = reduxForm(config)(
  FormApiServiceType
);
