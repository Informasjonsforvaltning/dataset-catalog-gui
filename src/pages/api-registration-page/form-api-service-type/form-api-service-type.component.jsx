import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import { Field } from 'redux-form';
import localization from '../../../services/localization';
import { Helptext } from '../../../components/helptext/helptext.component';
import SelectField from '../../../components/fields/field-select/field-select.component';

const apiServiceTypeSortOrder = {
  Kundeforhold: 0,
  Kontoopplysninger: 1
};

export const FormApiServiceType = ({ apiServiceTypeItems }) => (
  <form>
    <div className="form-group">
      <Helptext title={localization.schema.apiServiceType.title} />
      <Field
        name="serviceType"
        component={SelectField}
        items={_.sortBy(
          apiServiceTypeItems,
          item => apiServiceTypeSortOrder[item.code]
        )}
        valueKey="code"
        saveObject={false}
      />
    </div>
  </form>
);

FormApiServiceType.defaultProps = {
  apiServiceTypeItems: []
};

FormApiServiceType.propTypes = {
  apiServiceTypeItems: PropTypes.array
};
