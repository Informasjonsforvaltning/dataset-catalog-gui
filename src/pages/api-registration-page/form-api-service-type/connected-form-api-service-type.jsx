import { getFormSyncErrors } from 'redux-form';
import { connect } from 'react-redux';
import _ from 'lodash';
import { ConfiguredFormApiServiceType } from './configured-form-api-service-type';

const mapStateToProps = (state, ownProps) => {
  const { apiItem } = ownProps;
  return {
    syncErrors: getFormSyncErrors('apiServiceType')(state.form),
    initialValues: {
      serviceType: _.get(apiItem, 'serviceType')
    }
  };
};

export const ConnectedFormApiServiceType = connect(mapStateToProps)(
  ConfiguredFormApiServiceType
);
