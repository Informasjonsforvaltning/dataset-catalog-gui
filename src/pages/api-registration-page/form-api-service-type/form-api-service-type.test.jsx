import React from 'react';
import { shallow } from 'enzyme';
import { FormApiServiceType } from './form-api-service-type.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {};
  wrapper = shallow(<FormApiServiceType {...defaultProps} />);
});

test('should render FormApiServiceType, { renderLandingpage  correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
