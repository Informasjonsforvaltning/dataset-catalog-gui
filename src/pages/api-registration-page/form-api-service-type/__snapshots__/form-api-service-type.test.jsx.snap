// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `should render FormApiServiceType, { renderLandingpage  correctly 1`
] = `
<form>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      title="Tjenestetype (standard)"
    />
    <Field
      component={[Function]}
      items={Array []}
      name="serviceType"
      saveObject={false}
      valueKey="code"
    />
  </div>
</form>
`;
